horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -58,32 +64,403 @@ public class RunServiceTest extends BaseServiceTest {
+      String functionF = createFunction("sheep");
+      Transformer transformerF = createExampleTransformerWithJsonPath("tranF", schemaF, functionF, pathF);
+
+      Test exampleTest = createExampleTest(getTestName(info));
+      Test test = createTest(exampleTest);
+      addTransformer(test, transformerE, transformerF);
+
+      uploadRun(runWithValue(42.0d, schemaF),test.name);
+
+      DataSet dataSet = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+      JsonNode eventData = dataSet.data;
+      assertTrue(eventData.isArray());
+      assertEquals(1, eventData.size());
+      assertEquals(postFunctionSchemaUri(schemaF), eventData.get(0).path("$schema").asText());
+      assertEquals(42, eventData.get(0).path("outcome").path("sheep").intValue());
+   }
+
+   private void testTransformationWithoutMatch(TestInfo info, Schema schema, ObjectNode data) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+
+      NamedJsonPath firstMatch = createExampleNamedJsonPath("foo", "$.foo");
+      NamedJsonPath allMatches = createExampleNamedJsonPath("bar", "$.bar[*].x");
+      allMatches.array = true;
+      NamedJsonPath value = createExampleNamedJsonPath("value", "$.value");
+      NamedJsonPath values = createExampleNamedJsonPath("values", "$.values[*]");
+      values.array = true;
+
+      Transformer transformerNoFunc = createExampleTransformerWithJsonPath("noFunc", schema, null, firstMatch, allMatches);
+      Transformer transformerFunc = createExampleTransformerWithJsonPath("func", schema, "({foo, bar}) => ({ foo, bar })", firstMatch, allMatches);
+      Transformer transformerCombined = createExampleTransformerWithJsonPath("combined", schema, null, firstMatch, allMatches, value, values);
+
+      Test test = createTest(createExampleTest(getTestName(info)));
+      addTransformer(test, transformerNoFunc, transformerFunc, transformerCombined);
+      uploadRun(data, test.name);
+      DataSet dataset = dataSetQueue.poll(POLL_DURATION_SECONDS, TimeUnit.SECONDS);
+
+      assertNotNull(dataset);
+      assertTrue(dataset.data.isArray());
+      assertEquals(3, dataset.data.size());
+
+      JsonNode combined = StreamSupport.stream(dataset.data.spliterator(), false)
+            .filter(item -> item.get("value") != null).findFirst().orElseThrow(AssertionError::new);
+      assertEquals(42, combined.path("value").intValue());
+      assertTrue(combined.path("values").isArray());
+      assertEquals(3, combined.path("values").size());
+
+      JsonNode nodenoFunc = dataset.data.get(0);
+      JsonNode nodetransformerFunc = dataset.data.get(1);
+      JsonNode nodetransformerCombined = dataset.data.get(2);
+
+      assertEquals(3, nodenoFunc.size());
+      assertFalse(nodenoFunc.path("value").isMissingNode());
+      assertFalse(nodenoFunc.path("values").isMissingNode());
+      assertTrue(nodetransformerFunc.path("foo").isNull());
```

Comment:
You've removed the assertions about `foo` and `bar` in `nodenoFunc` and `nodetransformerCombined`